openedx/features/content_tagging/models.py

```diff
@@ -0,0 +1,121 @@
+"""
+Content Tagging models
+"""
+from django.db import models
+from django.utils.translation import gettext as _
+from opaque_keys import InvalidKeyError, OpaqueKey
+from opaque_keys.edx.keys import CourseKey, UsageKey
+from opaque_keys.edx.locator import LibraryLocatorV2, LibraryUsageLocatorV2
+from openedx_tagging.core.tagging.models import ObjectTag, Taxonomy
+from organizations.models import Organization
+
+
+class TaxonomyOrg(models.Model):
+    """
+    Represents the many-to-many relationship between Taxonomies and Organizations.
+
+    We keep this as a separate class from ContentTaxonomy so that class can remain a proxy for Taxonomy, keeping the
+    data models and usage simple.
+    """
+
+    class RelType(models.TextChoices):
+        OWNER = "OWN", _("owner")
+
+    taxonomy = models.ForeignKey(Taxonomy, on_delete=models.CASCADE)
+    org = models.ForeignKey(Organization, on_delete=models.CASCADE)
+    rel_type = models.CharField(
+        max_length=3,
+        choices=RelType.choices,
+        default=RelType.OWNER,
+    )
+
+    class Meta:
+        indexes = [
+            models.Index(fields=["taxonomy", "rel_type"]),
+            models.Index(fields=["taxonomy", "rel_type", "org"]),
+        ]
+
+    @classmethod
+    def get_owners(
+        cls,
+        taxonomy: Taxonomy,
+    ) -> models.QuerySet:
+        """
+        Returns the org "owners" on the given taxonomy.
+
+        If the returned list is empty, then the returned taxonomy is enabled for all organizations.
+        """
+        return cls.objects.filter(
+            taxonomy=taxonomy,
+            rel_type=cls.RelType.OWNER,
+        )
+
+    @classmethod
+    def is_owner(cls, taxonomy: Taxonomy, org_short_name: str) -> bool:
+        """
+        Returns True if there's an "owner" relationship between the given taxonomy and an org with the given short name.
+
+        Or, if no "owner" relationships exist for the given taxonomy, then all orgs are enabled for that taxonomy.
+        """
+        org_owners = cls.get_owners(taxonomy)
+        if not org_owners.exists():
+            return True
+        org_owners = org_owners.filter(org__short_name=org_short_name)
+        return org_owners.exists()
+
+
+class ContentTag(ObjectTag):
```

Comment:
Happy to call this `ContentObjectTag` instead if you think it's clearer?

Reply:
Done here: https://github.com/openedx/edx-platform/pull/32661/files#diff-8a644422ddd022128d13b6ca7c8b672a33d048d1816a80a80491246bdd76f607R69